goagen/gen_client/generator.go

```diff
@@ -674,8 +702,10 @@ func toString(name, target string, att *design.AttributeDefinition) string {
 			return fmt.Sprintf("%s := strconv.FormatBool(%s)", target, name)
 		case design.NumberKind:
 			return fmt.Sprintf("%s := strconv.FormatFloat(%s, 'f', -1, 64)", target, name)
-		case design.StringKind, design.DateTimeKind, design.UUIDKind:
+		case design.StringKind:
 			return fmt.Sprintf("%s := %s", target, name)
+		case design.DateTimeKind, design.UUIDKind:
+			return fmt.Sprintf("%s := %s.String()", target, strings.Replace(name, "*", "", -1)) // remove pointer if present
```

Comment:
the 'name' here is the ValueName from paramData. ValueName will have a pointer prefix to VarName if Param is Required.